lib/OpenQA/CLI/monitor.pm

```diff
@@ -0,0 +1,73 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::CLI::monitor;
+use OpenQA::Jobs::Constants;
+use Mojo::Base 'OpenQA::Command', -signatures;
+use Mojo::Util qw(encode getopt);
+
+has description => 'Monitors a set of jobs';
+has usage => sub { shift->extract_usage };
+
+sub _monitor_jobs ($self, $client, $poll_interval, $job_ids, $job_results) {
+    while (@$job_results < @$job_ids) {
+        my $job_id = $job_ids->[@$job_results];
+        my $tx = $client->build_tx(GET => $self->url_for("experimental/jobs/$job_id/status"), {});
+        my $res = $self->retry_tx($client, $tx);
+        return $res if $res != 0;
+        my $job = $tx->res->json;
+        my $job_state = $job->{state} // NONE;
+        if (OpenQA::Jobs::Constants::meta_state($job_state) eq OpenQA::Jobs::Constants::FINAL) {
+            push @$job_results, $job->{result} // NONE;
+            next;
+        }
+        print encode('UTF-8', "Job state of job ID $job_id: $job_state, waiting …\n");
+        sleep $poll_interval;
+    }
+}
+
+sub _compute_return_code ($self, $job_results) {
+    for my $job_result (@$job_results) {
+        return 2 unless OpenQA::Jobs::Constants::is_ok_result($job_result);
+    }
+    return 0;
+}
```

Comment:
These functions are exactly as they were before in `schedule.pm`.

Reply:
Fair enough, i'm still reviewing this as if it was a new feature.